### PR TITLE
[Tests-Only] Interactions for apps test added into apps test file

### PR DESCRIPTION
**Pull request by jasson99**
## Related Issue:
- #500

### CODE REVIEW

File: tests/appsTest.js [hunk 1]
```diff
@@ -15,10 +15,133 @@ fdescribe('Main: Currently testing apps management,', function () {
...
+          headers: validAuthHeaders
+        },
+        willRespondWith: {
+          status: 200,
+          headers: xmlResponseHeaders,
+          body: '<?xml version="1.0"?>\n' +
+            '<ocs>\n' +
+            ocsMeta('ok', '100') +
+            data +
+            '</ocs>'
```
Comment (swoichha) [inline tests/appsTest.js:114]: We can define this into a constant to make the code DRY as it is repeated elsewhere as well.

File: tests/appsTest.js [hunk 2]
```diff
@@ -131,6 +279,42 @@ fdescribe('Main: Currently testing apps management,', function () {
...
```
Comment (swoichha) [inline tests/appsTest.js:312]: same

**Review by swoichha — CHANGES_REQUESTED** [review]

File: tests/appsTest.js [hunk 3]
```diff
@@ -15,10 +15,133 @@ fdescribe('Main: Currently testing apps management,', function () {
...
+        for (const [key, value] of Object.entries(attributes[attribute].value)) {
+          data = data +
+            '  <element>\n' +
+            '   <key>' + utf8.encode(key) + '</key>\n' +
+            '   <app>someAppName</app>\n' +
+            '   <value>' + utf8.encode(value) + '</value>\n' +
+            '  </element>\n'
+        }
+        data = data + ' </data>'
+      } else if (attributes[attribute].attrExists === true) {
```
Comment (kiranparajuli589) [inline tests/appsTest.js:87]:
```suggestion
      } else if (attributes[attribute].attrExists) {
```

File: tests/appsTest.js [hunk 4]
```diff
@@ -15,10 +15,133 @@ fdescribe('Main: Currently testing apps management,', function () {
...
+        attrExists: true,
+        value: ''
+      }
+    }
+    for (const attribute in attributes) {
+      // default no data
+      let data = ' <data/>\n'
+
+      // no attributes specified, return all attributes
+      if (attribute === '') {
```
Comment (kiranparajuli589) [inline tests/appsTest.js:76]:
```suggestion
      if (!attribute) {
```

File: tests/appsTest.js (hunk 1 above)
Comment (jasson99) [reply tests/appsTest.js]: fixed

File: tests/appsTest.js (hunk 3 above)
Comment (jasson99) [reply tests/appsTest.js]: fixed

File: tests/appsTest.js (hunk 4 above)
Comment (jasson99) [reply tests/appsTest.js]: fixed

File: tests/appsTest.js [hunk 5]
```diff
@@ -9,16 +7,138 @@ fdescribe('Main: Currently testing apps management,', function () {
...
-  var testApp = 'someAppName'
-  var nonExistentApp = 'nonExistentApp' + timeRightNow
+  var nonExistentApp = 'nonExistentApp123'
 
   // PACT setup
   const Pact = require('@pact-foundation/pact-web')
   const provider = new Pact.PactWeb()
-  const { setGeneralInteractions } = require('./pactHelper.js')
+  const { setGeneralInteractions, ocsMeta } = require('./pactHelper.js')
+  const { validAuthHeaders, xmlResponseHeaders } = require('./pactHelper.js')
```
Comment (swoichha) [inline tests/appsTest.js:16]: You can do something like this
```suggestion
   const {
    setGeneralInteractions,
    ocsMeta,
    validAuthHeaders,
    xmlResponseHeaders
  } = require('./pactHelper.js')
```

**Review by HariBhandari07 — COMMENTED** [review]
I think it's nice to replace all `var`s with `let`s and `const`s as it is the modern way of js and it also makes the variable declaration in the PR more consistent.

File: tests/appsTest.js [hunk 6]
```diff
@@ -9,16 +7,137 @@ fdescribe('Main: Currently testing apps management,', function () {
...
   // TESTING CONFIGS
-  var testApp = 'someAppName'
-  var nonExistentApp = 'nonExistentApp' + timeRightNow
+  var nonExistentApp = 'nonExistentApp123'
 
   // PACT setup
   const Pact = require('@pact-foundation/pact-web')
   const provider = new Pact.PactWeb()
-  const { setGeneralInteractions } = require('./pactHelper.js')
+  const { validAuthHeaders, xmlResponseHeaders, setGeneralInteractions, ocsMeta } = require('./pactHelper.js')
```
Comment (jasson99) [inline tests/appsTest.js:15]: .

File: tests/appsTest.js (hunk 5 above)
Comment (jasson99) [reply tests/appsTest.js:16]: I prefer as it is now. I don't want to add extra lines to import the same thing.

**Review by swoichha — COMMENTED** [review]
LGTM :+1:

**Review by swoichha — APPROVED** [review]

**Review by individual-it — APPROVED** [review]

**Pull request merged**